Display more information in tracker tab

ngosang opened the pull request:
Before:
![](http://i.imgur.com/DobUmPn.png)

After (only with libtorrent >= 1.0.0):
![](http://i.imgur.com/cB5jhfw.png)

ngosang commented:
Looks like there is a problem with Dowloaded stats (aren't retrieved from some trackers) but should work when get fixed. https://code.google.com/p/libtorrent/issues/detail?id=755

ngosang commented:
I will rebase if someone thinks it's useful.

DoumanAsh commented:
I think no one would be against it and 'Seeds' is useful.
What is received?

Downloaded ain't really useful i think, but well no harm

ngosang commented:
The tracker doesn't send all seeds/peers, just 50/200... random depend on
tracker config. This saves tracker bandwidth.

DoumanAsh commented:
Hm... well it sounds strange to me... But if it is so then ok.
Regardless i thin if anything should be added, it is only number of Seeds.
Others: Downloaded and Received, i cannot imagine who would need that from tracker

ngosang commented:
> Hm... well it sounds strange to me... But if it is so then ok.

It's how the trackers work. Look at demonii in the screenshot, has more than 29000 seeds + 11000 peers = 40000 users. The tracker don't give you a list with 40000 ips/ports, it only gives you 200 ips from that list and the next time you announce to the tracker it gives you other 200 peers and so on...

> Regardless i thin if anything should be added, it is only number of Seeds.

In the first screenshot the column "Peers" is the same as "Received" in the second screenshot (the numbers are lower than 200 because the screenshot is from a different torrent). I renamed "Peers" to "Received" and added 3 new columns: "Seeds", "Peers", "Downloaded".

ngosang commented:
Rebased and updated the screenshots in the first message.

Nemo-qB commented:
+1

Nice, would like to see this implented into qBittorrent.

chrishirst commented:
> Others: Downloaded and Received, i cannot imagine who would need that from tracker

Actually those metrics would be FAR more accurate for calculating and displaying the 'share ratio' for private tracker torrents than the current method is,  as it would be using the **exact same** metrics as the indexer site is using. So it would be far more useful to display the 'ratio' in the trackers tab and drop the task list ratio information.

thalieht commented:
I like this too :thumbsup:

Chocobo1 commented:
I am not sure this make sense: I would like to see the value `next_announce` added.
It's the time till the next announce.

ngosang commented:
> I would like to see the value next_announce added.

I tried to add but it works wrong. Sometimes the next_announce value doesn't make sense...

http://www.libtorrent.org/reference-Core.html#announce_entry
```
   ptime next_announce;
   ptime min_announce;
   int scrape_incomplete;
   int scrape_complete;
   int scrape_downloaded;
```
You can try to investigate that is what is wrong.

Chocobo1 commented:
What about using `int next_announce_in()`?

ngosang commented:
> What about using int next_announce_in()?

I think I tried also but I don't remember.

ngosang commented:
@sledgehammer999 when you have time could you read the entire conversation? All clients have this feature but I can't implement it properly because a missing feature in libtorrent. Maybe you can talk with arvidn.

LordNyriox commented:
@ngosang Might it be possible for you to split your commit into two, one commit containing all the changes which are compatible with current libtorrent (which could be merged separately), while the other commit contains those changes not currently compatible with libtorrent features?  

Not that I have any clue which is which.  :\  

Thanks, 
~ Nyriox

sledgehammer999 commented:
arvidn posted in the mailing list that he's going to release 1.1.x soon. And asked if something needs attention. I pointed him to https://github.com/arvidn/libtorrent/issues/29 again. I don't know if he'll decide to implement it.

As for the wrong next announce time: We should report it as a bug.

LordNyriox commented:
@ngosang Do you think you can update this PR (with some king of `if`/`else` set to require Libtorrent 1.1.x for the changes) based on the manual scrape fixes in [https://github.com/arvidn/libtorrent/pull/433]?  I believe those changes in Libtorrent might be able to fix your previously mentioned problems. 

Thanks, 
~ Nyriox

LordNyriox commented on src/gui/properties/trackerlist.cpp:
```diff
@@ -284,8 +287,18 @@ void TrackerList::loadTrackers() {
     }
+    item->setText(COL_RECEIVED, QString::number(data.numPeers));
+
+#if LIBTORRENT_VERSION_NUM >= 10000
```
@ngosang Replace `>= 10000` with `>= 10100` here.  The scrape problem is fixed in Libtorrent v1.1 (https://github.com/arvidn/libtorrent/pull/433).  

Thanks, 
~ Nyriox

LordNyriox commented:
@ngosang:  Supposedly the Libtorrent scrape issue is fixed in Libtorrent 1.1.0.  Can you rebase this PR and test whether this works with that version?  

Thanks, 
~ Nyriox

LordNyriox commented:
@sledgehammer999:  I'm not seeing much activity on this PR right now.  Supposedly, however, the scrape issue with Libtorrent was fixed in version 1.1, so I believe if you modify the required Libtorrent version for this feature to that, you shouldn't have any problems.  

Could you test this with Libtorrent 1.1, just to make sure it works as intended?

ngosang commented:
@sledgehammer999 rebased. Could you test it a bit? I think could be merged even if the libtorrent bug is not fixed.

LordNyriox commented:
@ngosang:  Did you change the libtorrent requirement to 1.1 (instead of 1.0), as I suggested?  The Libtorrent bug was supposedly fixed in that branch, not 1.0.  

Thanks for rebasing.

ngosang commented:
> Did you change the libtorrent requirement to 1.1 (instead of 1.0), as I suggested? 

No. I'm using 1.0.9 and it partially works. The only minor issue is the Completed column sometimes is 0.

LordNyriox commented:
@ngosang:  All right then.  Thanks.

LordNyriox commented:
@sledgehammer999:  Can you go ahead and merge this?  

Thanks.

DrKittens commented:
+1 would love to see this.
Makes the trackers page less confusing/actually provide useful scrape information..

LordNyriox commented:
@sledgehammer999:  Can you at least add this to the `3.3.5` milestone, if you are not ready to merge this yet?  

Thanks.

LordNyriox commented:
@ngosang:  Do you think you can rebase this PR?  Github isn't reporting conflicts now, but with all the major code changes coming down the pike it will not be long before Github does.  

Thank you.

ngosang commented:
This could be merged as it's.

LordNyriox commented:
@ngosang:  To be clear, there's a huge PR by @glassez that's currently queued up, and it changes a whole lot of code, all over the place, to push further the transition to `SettingsStorage`.  

That same PR also includes a great many optimizations to smooth over porting to Libtorrent 1.1.  

I really have no idea what such a huge PR might do to your little one, so I'm just informing you to be prepared to rebase in the near future.  

Thank you for your patience.

DrKittens commented:
Casual Bump, this is still functional after the aforementioned glasses change, i'd love to see it merged still.

LordNyriox commented:
@sledgehammer999:  Since this PR has no merge conflicts ATM, can it be merged?

LordNyriox commented:
@sledgehammer999:  Can you please merge this PR?  It's relatively old, but since you just finished a release, there should be plenty of time to test it and fix any bugs.  

Thanks again!

thalieht commented:
@ngosang here's a little patch that:

- Separates seeds from peers for DHT PeX LSD
- Changes zeroed  values that can't change to empty strings (if you see a zero you expect that it can change).
- Some coding style.

If you want, take whatever you want from it :)

https://gist.github.com/thalieht/14ae3106ae89f82d71a2a2c9f9ad29ed

About the Downloaded column: I tested a torrent with all the trackers from your trackerslist with libtorrent 1.1.1 and all udp trackers showed zero.

zywo commented:
17 months and not merged yet.

sledgehammer999 commented:
Thank you and sorry for the long delay.
